### DISCUSSION TITLE
Add WinUI control to control library

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#10562] [Q&A] [answered]
I am following [these instructions](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html) to create a control library.

> Choose the template flavor based on your library's flavor: UWP or WinUI. If your project uses the Uno.UI NuGet package, it's UWP otherwise WinUI 3 if it uses Uno.WinUI.

I'm not really sure what "your project" means since dependencies are per-platform.  My intent is to use WinUI 3 and when I look at Dependencies/netstandard2.0/packages I see a reference to Uno.WinUI 4.6.19 so I presume I'm good there.

Problem is when I "Right-click on the project library, then Add, New Item" per the doc I only see UWP templates.  The words "Choose the template flavor..." make me think I should see WinUI as well as UWP (??).

![Capture](https://user-images.githubusercontent.com/16885210/203849533-147d77f4-213b-411e-93e1-3f53c719cac2.PNG)

When I choose Custom Contro (Uno Platform UWP) a control is added to the project however references to Windows.UI.Xaml.Controls, Windows.UI.Xaml.Data, etc. are not found the project does not build.

How can I create a WinUI control in the project?

I am targeting .net 7 [per my question here](https://github.com/unoplatform/uno/discussions/10529).
Using VS 17.4.1

**Comment by jeromelaban** [comment, marked as answer]
We're currently adjust those templates so that they fit better with the WinUI/WinAppSDK scenario. In the meantime, you can create a UWP one, then rename anything `Windows.UI.Xaml` to `Microsoft.UI.Xaml` in both the `.xaml` and `.xaml.cs` files.
